### PR TITLE
Spec MSC2285: Private read receipts

### PR DESCRIPTION
**Pull request by turt2live**
Markdown (used as reference due to post-FCP edits): https://github.com/matrix-org/matrix-spec-proposals/blob/main/proposals/2285-hidden-read-receipts.md

MSC: https://github.com/matrix-org/matrix-spec-proposals/pull/2285

**Requires https://github.com/matrix-org/matrix-spec-proposals/pull/3875**

**Reviewer**: This is best reviewed commit-by-commit.

Preview: https://pr1216--matrix-spec-previews.netlify.app

### CODE REVIEW

**Review by SimonBrandner — APPROVED** [review]
LGTM - thank you!

File: data/api/client-server/receipts.yaml [hunk 1]
```diff
@@ -49,7 +50,10 @@ paths:
           description: The type of receipt to send.
           required: true
           x-example: "m.read"
-          enum: ["m.read"]
+          x-changedInMatrixVersion:
+            1.4: |
+              Allow `m.read.private` receipts.
```
Comment (clokep) [inline data/api/client-server/receipts.yaml:55]: `/receipts` also now allows `m.fully_read`.

**Comment by turt2live** [comment]
while we appreciate the support, approving PRs like this does not help them move faster :)

File: content/client-server-api/modules/push.md [hunk 2]
```diff
@@ -107,7 +107,19 @@ determined by the push rules which apply to an event.
 
 When the user updates their read receipt (either by using the API or by
 sending an event), notifications prior to and including that event MUST
-be marked as read.
+be marked as read. Note that users can send both an `m.read` and
+`m.read.private` receipt, both of which are capable of clearing notifications.
+
+If the user has both `m.read` and `m.read.private` set in the room then
+the receipt which is more recent/ahead must be used to determine where
+the user has read up to. For example, given an ordered set of events A,
```
Comment (ara4n) [inline content/client-server-api/modules/push.md:115]: ordered by what metric?

File: content/client-server-api/modules/receipts.md [hunk 3]
```diff
@@ -4,10 +4,14 @@ type: module
...
+{{< changed-in v="1.4" >}} Added private read receipts.
+
 This module adds in support for receipts. These receipts are a form of
-acknowledgement of an event. This module defines a single
-acknowledgement: `m.read` which indicates that the user has read up to a
-given event.
+acknowledgement of an event. This module defines the `m.read` receipt
+for indicating that the user has read up to a given event, and `m.read.private`
+to achieve the same purpose without any other user being aware. Primarily,
+`m.read.private` is meant to clear [notifications](#receiving-notifications)
```
Comment (ara4n) [inline content/client-server-api/modules/receipts.md:13]:
```suggestion
`m.read.private` is intended to clear [notifications](#receiving-notifications)
```
minor phrasing quibble

File: content/client-server-api/modules/receipts.md [hunk 4]
```diff
@@ -59,12 +63,36 @@ following HTTP APIs.
 
 {{% http-api spec="client-server" api="receipts" %}}
 
+##### Private read receipts
+
+{{% added-in v="1.4" %}}
+
+Some users would like to clear their [notification counts](#receiving-notifications),
```
Comment (ara4n) [inline content/client-server-api/modules/receipts.md:70]:
```suggestion
Some users would like to mark a room as read, clearing their [notification counts](#receiving-notifications),
```

File: content/client-server-api/modules/push.md (hunk 2 above)
Comment (turt2live) [reply content/client-server-api/modules/push.md:115]:
```suggestion
the user has read up to. For example, given an oldest-first set of events A,
```

**Review by ara4n — APPROVED** [review]
lgtm with minor quibbles. very pleasing to see what a simple diff it is in the end to the spec itself :)

**Pull request merged**